source/MaterialXTest/ShaderGen.cpp

```diff
@@ -160,60 +126,62 @@ enum LightType
     DIRECTIONAL = 4,
 };
 
+void createLights(mx::DocumentPtr doc, mx::HwLightHandler& lightHandler)
+{
+    for (mx::NodePtr node : doc->getNodes())
+    {
+        if (node->getType() == LIGHT_SHADER)
+        {
+            mx::LightSourcePtr light;
+            if (node->getCategory() == DIRECTIONAL_LIGHT)
+            {
+                light = lightHandler.createLightSource(DIRECTIONAL);
+            }
+            else if (node->getCategory() == POINT_LIGHT)
+            {
+                light = lightHandler.createLightSource(POINT);
+            }
+            else if (node->getCategory() == SPOT_LIGHT)
+            {
+                light = lightHandler.createLightSource(SPOT);
+            }
+            else if (node->getCategory() == LIGHT_COMPOUND)
+            {
+                light = lightHandler.createLightSource(LIGHT_COMPOUND_ID);
+            }
+            if (light)
+            {
+                for (mx::InputPtr input : node->getInputs())
+                {
+                    light->setParameter(input->getName(), input->getValue());
+                }
+            }
+        }
+    }
+}
+
 void createLightRig(mx::DocumentPtr doc, mx::HwLightHandler& lightHandler, mx::HwShaderGenerator& shadergen, const mx::GenOptions& options)
 {
     // Create a custom light shader by a graph compound
-    createLightCompoundExample(doc);
-
     mx::NodeDefPtr dirLightNodeDef = doc->getNodeDef("ND_directionallight");
```

Comment:
You've still left this code here. It should all be removed and added dynamically if a nodedef of a given type is found above. (add getNodeDef() and addLightShader() lines).